server/hwapi/external/c3/api.py

```diff
@@ -0,0 +1,151 @@
+                        .first()
+                    )
+                    # Remove Inc/Inc. from vendor name to avoid duplicate vendors
+                    if bios_vendor is None and (
+                        public_cert.bios.vendor.endswith("Inc")
+                        or public_cert.bios.vendor.endswith("Inc.")
+                    ):
+                        bios_vendor_name = (
+                            public_cert.bios.vendor[
+                                : public_cert.bios.vendor.rfind("Inc")
+                            ]
+                            .replace(",", "")
+                            .strip()
+                        )
+                        bios_vendor = (
+                            self.db.query(models.Vendor)
+                            .filter(models.Vendor.name.ilike(bios_vendor_name))
+                            .first()
+                        )
+                    if bios_vendor is None:
+                        bios_vendor = get_or_create(
+                            self.db, models.Vendor, name=public_cert.bios.vendor
+                        )
+                    bios = get_or_create(
+                        self.db,
+                        models.Bios,
+                        revision=public_cert.firmware_revision,
+                        version=(
+                            public_cert.bios.version
+                            if public_cert.bios.version
+                            else public_cert.bios.name
+                        ),
+                        vendor=bios_vendor,
+                    )
+                release = get_or_create(
+                    self.db,
+                    models.Release,
+                    codename=public_cert.release.codename,
+                    release=public_cert.release.release,
+                    release_date=public_cert.release.release_date,
+                    i_version=public_cert.release.i_version,
+                    supported_until=public_cert.release.supported_until,
+                )
+                certificate = get_or_create(
+                    self.db,
+                    models.Certificate,
+                    name=public_cert.name,
+                    completed=public_cert.completed,
+                    created_at=public_cert.created_at,
+                    machine=machine,
+                    release=release,
+                )
+                get_or_create(
+                    self.db,
+                    models.Report,
+                    architecture=public_cert.architecture,
+                    kernel=kernel,
+                    bios=bios,
+                    certificate=certificate,
+                )
+            except IntegrityError:
+                logging.error(
+                    "Got an error while importing certificates", exc_info=True
+                )
+                self.db.rollback()
+                continue
```

Comment:
what does the rollback do here? given that there is a continue after it. Would it rollback all objects added so far even ones that had no errors?